components/eventing-controller/pkg/backend/nats/jetstream/jetstream_test.go

```diff
@@ -1476,3 +1477,103 @@ func TestSubscriptionSubjectIdentifierNamespacedName(t *testing.T) {
 		})
 	}
 }
+
+// TestJetStream_NoNATSSubscription tests if the error is being triggered
+// when expected in-memory representation of NATS subscriptions are missing.
+func TestJetStream_NoNATSSubscription(t *testing.T) {
+	// given
+	testEnvironment := setupTestEnvironment(t)
+	jsBackend := testEnvironment.jsBackend
+	defer testEnvironment.natsServer.Shutdown()
+	defer testEnvironment.jsClient.natsConn.Close()
+	initErr := jsBackend.Initialize(nil)
+	require.NoError(t, initErr)
+
+	// create New Subscriber
+	subscriber := evtesting.NewSubscriber()
+	defer subscriber.Shutdown()
+	require.True(t, subscriber.IsRunning())
+
+	testCases := []struct {
+		name                            string
+		subOpts                         []evtesting.SubscriptionOpt
+		givenManuallyDeleteSubscription bool
+		givenFilterToDelete             string
+		wantNatsSubsLen                 int
+		wantErr                         bool
+		wantErrText                     string
+	}{
+		{
+			name: "No error should happen, when there are no filters",
+			subOpts: []evtesting.SubscriptionOpt{
+				evtesting.WithSinkURL(subscriber.SinkURL),
+				evtesting.WithEmptyFilter(),
+				evtesting.WithStatusConfig(env.DefaultSubscriptionConfig{MaxInFlightMessages: 10}),
+			},
+			givenManuallyDeleteSubscription: false,
+			wantNatsSubsLen:                 0,
+			wantErr:                         false,
+		},
+		{
+			name: "No error should happen when NO in-memory representation of the NATS sub disappear",
+			subOpts: []evtesting.SubscriptionOpt{
+				evtesting.WithFilter("", evtesting.OrderCreatedEventType),
+				evtesting.WithFilter("", evtesting.OrderCreatedEventType+"2"),
+				evtesting.WithStatusConfig(env.DefaultSubscriptionConfig{MaxInFlightMessages: 10}),
+			},
+			givenManuallyDeleteSubscription: false,
+			wantNatsSubsLen:                 2,
+			wantErr:                         false,
+		},
+		{
+			name: "An error is expected, when we manually delete an in-memory representation of the NATS subscription",
+			subOpts: []evtesting.SubscriptionOpt{
+				evtesting.WithFilter("", evtesting.OrderCreatedEventType),
+				evtesting.WithFilter("", evtesting.OrderCreatedEventType+"2"),
```

Comment:
How about creating a new variable in evtesting as `NewOrderCreatedEventType`?

Reply:
If it's a nit-pick, I would like to leave it as is

Reply:
Its just that `evtesting.OrderCreatedEventType+"2"` has been used multiple times now and it makes sense to have its own variable.